oonib/report/handlers.py

```diff
@@ -402,6 +408,37 @@ def get(self):
     def post(self, report_id):
         closeReport(report_id)
 
+
+class NewMeasurementHandlerFile(NewReportHandlerFile):
+    """ Implements the /measurement endpoint.
+
+        See https://github.com/ooni/spec/blob/master/backends/bk-003-collector.md#34-submitting-single-measurements-using-a-single-api-call """
+
+    def post(self):
+        measurement = json.loads(self.request.body)
+        reportID = self.createReport({
+            'software_name': str(measurement['software_name']),
+            'software_version': str(measurement['software_version']),
+            'probe_asn': str(measurement['probe_asn']),
+            'probe_cc': str(measurement['probe_cc']),
+            'test_name': str(measurement['test_name']),
+            'test_version': str(measurement['test_version']),
+            'input_hashes': measurement.get('input_hashes', []),
+            'test_start_time': str(measurement['test_start_time']),
+            'data_format_version': str(
+                measurement.get('data_format_version', '0.2.0')
+            ),
+            'format': 'json',
+            'content': self.request.body,
+        })['report_id']
+        closeReport(reportID)
+        # TODO(bassosimone): the specification says that we should also
+        # return a measurement ID. Yet, it's not possible to add this
+        # functionality in here easily IMHO. So, should I also change the
+        # spec to say "SHOULD return a measurement ID"?
```

Comment:
I don't think we should change that as it's pretty important that we start stamping the measurement_id inside of the backend ASAP.

I am of the opinion that we should roll out this single measurement API using the new ooni/collector code which already supports measurement_id stamping.

Reply:
Actually we don't use the measurement_id as per the spec in https://github.com/ooni/pipeline/blob/master/docs/ooni-uuid.md#collector-stamped-ooid inside of the ooni/collector either.

At this point I am not sure what is the best way forward. I though think that once we start stamping the `measurement_id` in any collector the spec ought to say that it is strict requirement as this is a very important aspect of the collector which should not be optional.

Reply:
"The legacy implementation is allowed to return an empty string here for backwards compatibility, however new implementations MUST return a valid OOID."

Reply:
> I am of the opinion that we should roll out this single measurement API using the new ooni/collector code which already supports measurement_id stamping.

I disagree here. I'd like to deploy this new code in production when approved / next week, so the next iteration of mobile devices already takes advantage of this. I'd also like to deploy the new codebase as `d.collector.ooni.io` next week, without announcing it via bouncer, so we can start playing with it in our integration and automated tests.